File: Clearance/Models/PeriodicTask.swift
//
//  PeriodicTask.swift
//  Clearance
//

import Foundation

/// A recurring task that resets at the start of each week or month.
/// Persisted via PeriodicTaskStore (UserDefaults), not SwiftData.
struct PeriodicTask: Identifiable, Codable, Hashable, Sendable {
    let id: UUID
    var title: String
    var emoji: String
    var recurrence: Recurrence
    var isCompleted: Bool
    var completedDate: Date?

    init(id: UUID = UUID(), title: String, emoji: String = "📋", recurrence: Recurrence) {
        self.id = id
        self.title = title
        self.emoji = emoji
        self.recurrence = recurrence
        self.isCompleted = false
        self.completedDate = nil
    }
}

File: android/app/src/main/java/com/krya1012/clearance/data/PeriodicTask.kt
package com.krya1012.clearance.data

import kotlinx.serialization.Serializable
import java.util.UUID

/**
 * A recurring task that resets at the start of each week or month.
 * Persisted via PeriodicTaskStore (DataStore), not Room.
 */
@Serializable
data class PeriodicTask(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val emoji: String = "📋",
    val recurrence: Recurrence,
    val isCompleted: Boolean = false,
    /** Epoch millis of the completion, or null if not completed. */
    val completedDateMillis: Long? = null,
)
